src/libsync/capabilities.cpp

```diff
@@ -150,4 +150,15 @@ QString Capabilities::invalidFilenameRegex() const
 {
     return _capabilities["dav"].toMap()["invalidFilenameRegex"].toString();
 }
+
+QString Capabilities::getWebSocketUrl() const
+{
+    if (!_capabilities.contains("websocket")) {
+        return "";
```

Comment:
`return QString()`

(Otherwise you make a useless conversion from char*)

Also later.